pkg/block/blocktest/multipart_suite.go

```diff
@@ -304,24 +303,24 @@ func requireEqualBigByteSlice(t *testing.T, exp, actual []byte) {
 	}
 
 	for i := 0; i < sliceCount; i++ {
-		var start = i * sliceLen
-		var end = min((i+1)*sliceLen, len(exp)-1)
+		start := i * sliceLen
+		end := min((i+1)*sliceLen, len(exp)-1)
 
-		var expSlice = exp[start:end]
-		var actualSlice = actual[start:end]
+		expSlice := exp[start:end]
+		actualSlice := actual[start:end]
 		require.Equalf(t, expSlice, actualSlice, "Failed on slice "+strconv.Itoa(i+1)+"/"+strconv.Itoa(sliceCount))
```

Comment:
I reckon we may as well add the true byte offset here.

Reply:
?

Reply:
updated to check length and use deep.Equal